substrate/client/consensus/beefy/src/lib.rs

```diff
@@ -118,50 +119,62 @@ where
 /// Links between the block importer, the background voter and the RPC layer,
 /// to be used by the voter.
 #[derive(Clone)]
-pub struct BeefyVoterLinks<B: Block> {
+pub struct BeefyVoterLinks<B: Block, AuthorityId: AuthorityIdBound>
+where
+	<AuthorityId as RuntimeAppPublic>::Signature: Send + Sync,
```

Comment:
I've not investigated deeply. But seems that this `Send + Sync` requirement is quite omnipresent in this PR. Do you see a simple way to not explicitly specify it

Reply:
This is what we were struggling with together that night in October. Logically I should be able to add a `<Self as RuntimeAppPublic>::Signature: Send + Sync`, to `AuthorityIdBound` but adding that has no effect and I still have to add the bound to each function which has an `AuthorityIdBound` parameter. It seems a Rust bug to me. If could file a rust bug if it seems a bug to you as well. But I couldn't found any other way around it.

Reply:
I don't know if this is a rust bug. I think it's a grey area. There is a discussion here that seems related to something similar: https://github.com/rust-lang/rust/issues/20671 . A workaround for this would be something like:

```
pub trait AuthorityIdBound:
	Ord
	+ AppPublic
	+ Display
	+ BeefyAuthorityId<BeefySignatureHasher, Signature = Self::BoundedSignature>
{
	type BoundedSignature: Debug + Eq + PartialEq + Clone + TypeInfo + Codec + Send + Sync;
}
```

Reply:
I think you can just:
1. add `Send + Sync` as a bound to `RuntimeAppPublic::Signature` [here](https://github.com/w3f/polkadot-sdk/blob/f75e3a21ce9d85f889410886522f06eedd27dad3/substrate/primitives/application-crypto/src/traits.rs#L124)
2. add `Send + Sync` as a bound [here](https://github.com/w3f/polkadot-sdk/blob/f75e3a21ce9d85f889410886522f06eedd27dad3/substrate/primitives/application-crypto/src/traits.rs#L156) as well.

Then you should be able to get rid of all these send+sync for the `Signature`.
In general makes sense for a signature to be required to be send+sync

Reply:
I got rid of all of `RuntimeAppPublic::Signature: Sync + Send` s using @serban300 suggestion :100: